SNAP-1257

snappy-sachin opened the pull request:
## What changes were proposed in this pull request?

1. Adding SnappyData Product documentation link on UI.
2. Fixes for SnappyData Product version not displayed issue.


## How was this patch tested?

Tested Manually

sumwale commented:
@soubhik-c @kneeraj @suranjan please review

suranjan commented on core/src/main/scala/org/apache/spark/util/Utils.scala:
```diff
@@ -2230,7 +2232,34 @@ private[spark] object Utils extends Logging {
+  def tempFileWith(parent: String, prefix: String): File = {
+    var temp: File = null
+    var i = 0
+    while (i < 50) {
```
Why 50 times? can't have till we get a file that doesn't exist.

sumwale replied:
Ok

suranjan commented:
For my understanding, if this is for performance, I couldn't understand why new one is faster? While loop in previous version?

sumwale commented:
In Spark version there is no while loop. This is a possible bug in case there is a file name clash but very rare.

As stated in the issue, the new one is faster because it uses java.util.Random rather than the inbuilt one (UUID.randomUUID) which uses java.util.SecureRandom. Latter is slow and can potentially block for indefinite periods since it depends on collection of genuine random data (rather than pseudo-random numbers as used by java.util.Random). In perf analysis it frequently shows up as having a decent share of time. This level of security is not required for file names since the name will retry if present. Please see the issue for more details.

suranjan commented:
Ok. Looks good.

snappy-sachin committed:
SNAP-1257
1. Adding SnappyData Product documentation link on UI.
2. Fixes for SnappyData Product version not displayed issue.

snappy-sachin committed:
Merge branch 'snappy/branch-2.0' into SNAP-1257

snappy-sachin committed:
SNAP-1257: Renamed SnappyData Guide link as Docs.
